email_fetcher_service.py

```diff
@@ -0,0 +1,90 @@
+from googleapiclient.discovery import build
+from google.oauth2.credentials import Credentials
+from typing import Optional, List, Dict
+import json
+from sqlalchemy.exc import SQLAlchemyError
+
+from session_manager import get_db_session
+from models import UserToken
+
+from email_storage_service import EmailStorageService, FullMessage
+from google_auth_Manager import AuthManager
+from auth_utility import get_user_email
+
+
+class EmailFetcherService:
+    def __init__(self, cred_details: Credentials):
+        self.cred_details = cred_details
+        self.service = build("gmail", "v1", credentials=cred_details)
+        self.user_id = "me"
+        self.next_page_token: Optional[str] = None
+
+    def _build_query(
```

Comment:
Check and update doc string and typing in every places.

Reply:
Will be handled in https://github.com/SAGNICKMAJUMDAR/Filter-Email/issues/9?issue=SAGNICKMAJUMDAR%7CFilter-Email%7C11